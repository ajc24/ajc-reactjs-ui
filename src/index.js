export { default as Footer } from './ui-elements/Footer';
export { default as Header } from './ui-elements/Header';
export { default as HeaderLogo } from './ui-elements/HeaderLogo';
export { default as HeaderTitleAndSubtitle } from './ui-elements/HeaderTitleAndSubtitle';
export { default as MainContentPageTitle } from './ui-elements/MainContentPageTitle';
export { default as MainContentPanel } from './ui-elements/MainContentPanel';
export { default as MainContentText } from './ui-elements/MainContentText';
export { default as Menu } from './ui-elements/Menu';
export { default as Screenshot } from './ui-elements/Screenshot';
export { default as SmallMenuIcon } from './ui-elements/SmallMenuIcon';
export { default as SmallRightArrowIcon } from './ui-elements/SmallRightArrowIcon';
