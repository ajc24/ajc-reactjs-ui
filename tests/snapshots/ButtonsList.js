import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { ButtonsList } from '../../src';

describe('Buttons List', () => {
  const singleButtonList = [
    {
      id: 'test-single-button-id',
      onClick: jest.fn(),
      title: 'Test Default Single Grey Button',
      type: 'button',
    },
  ];
  const singleSubmitButtonList = [
    {
      id: 'test-single-submit-button-id',
      onClick: jest.fn(),
      title: 'Test Submit Button',
      type: 'submit',
    },
  ];
  const twoButtonsList = [
    {
      id: 'test-button-1-id',
      onClick: jest.fn(),
      title: 'Test Button 1',
      type: 'button',
    },
    {
      id: 'test-button-2-id',
      onClick: jest.fn(),
      title: 'Test Button 2',
      type: 'button',
    },
  ];
  describe('Default props and rendering - Single grey button', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ButtonsList buttonsList={singleButtonList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Two center aligned red buttons', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ButtonsList colour="red" alignment="center-align" buttonsList={twoButtonsList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Single right aligned grey submit button', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ButtonsList colour="grey" alignment="right-align" buttonsList={singleSubmitButtonList} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
