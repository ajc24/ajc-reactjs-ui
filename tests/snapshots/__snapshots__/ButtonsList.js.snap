// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons List Default props and rendering - Single grey button verifies the snapshot for the component 1`] = `
<div
  className="ajc-buttons-list-container ajc-buttons-list-container-left-align ajc-default"
>
  <button
    className="ajc-button ajc-default ajc-button-grey"
    id="test-single-button-id"
    onClick={[MockFunction]}
    type="button"
  >
    Test Default Single Grey Button
  </button>
</div>
`;

exports[`Buttons List Transferred props and rendering - Single right aligned grey submit button verifies the snapshot for the component 1`] = `
<div
  className="ajc-buttons-list-container ajc-buttons-list-container-right-align ajc-default"
>
  <button
    className="ajc-button ajc-default ajc-button-grey"
    id="test-single-submit-button-id"
    onClick={[MockFunction]}
    type="submit"
  >
    Test Submit Button
  </button>
</div>
`;

exports[`Buttons List Transferred props and rendering - Two center aligned red buttons verifies the snapshot for the component 1`] = `
<div
  className="ajc-buttons-list-container ajc-buttons-list-container-center-align ajc-default"
>
  <button
    className="ajc-button ajc-default ajc-button-red ajc-button-right-spacing"
    id="test-button-1-id"
    onClick={[MockFunction]}
    type="button"
  >
    Test Button 1
  </button>
  <button
    className="ajc-button ajc-default ajc-button-red"
    id="test-button-2-id"
    onClick={[MockFunction]}
    type="button"
  >
    Test Button 2
  </button>
</div>
`;
